### PR TITLE
Add callbackAfterInit methods for executing functions after the app is initialized

### PR DESCRIPTION
**Pull request by vinothvino42**
Ticket: #958

In this PR, I added an `addCallbackAfterInitialization` and `getCallbacksAfterInitialization`. Also updated the ensemble app to execute all the registered functions

### CODE REVIEW

**Comment by vusters** [comment]
@vinothvino42 let's do this right and remove the hacky delay. Try this:
1. Add `Ensemble().addCallbackAfterInitialization(func)`. In your deep link, call this and pass the function to navigate the screen. The function will just add the function to the list of functions it stores internally.
2. in EnsembleApp initState(), after the first frame look at Ensemble().getCallbacksAfterInitialization() and execute them in order. Make sure to clear each func out before you execute, in case some error causes the function to stay there indefinintely and hose the app.

**Comment by vinothvino42** [comment]
@vusters Can you review it again?

File: lib/ensemble_app.dart [hunk 1]
```diff
@@ -117,10 +117,38 @@ class EnsembleAppState extends State<EnsembleApp> with WidgetsBindingObserver {
       Workmanager().initialize(callbackDispatcher, isInDebugMode: false);
       initDeepLink(AppLifecycleState.resumed);
     }
+
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      List<int?> functionIds = [];
+      final callbacks = Ensemble().getCallbacksAfterInitialization();
+      for (final Map<String, Object?> callback in callbacks) {
+        final id = callback['id'] as int?;
```
Comment (vusters) [inline lib/ensemble_app.dart]: see my comment above, it'll simplify this code.

File: lib/ensemble_app.dart [hunk 2]
```diff
@@ -117,10 +117,38 @@ class EnsembleAppState extends State<EnsembleApp> with WidgetsBindingObserver {
...
+
+        if (method != null) {
+          await Function.apply(method, positionalPayloads, namedParams);
+          functionIds.add(id);
+        }
+      }
+
+      // Looping function id to remove the functions in the callback object with the id
+      for (final id in functionIds) {
+        callbacks.removeWhere((element) => element['id'] == id);
```
Comment (vusters) [inline lib/ensemble_app.dart]: Please read my original comment for this task. Either you put function execution inside try/catch, or you remove the item then execute (with a try catch). This way if the function throws an exception, our app is not hosed permanently. Remmeber this is what the app will do on start-up - a bad function will mess the app, regardless if you restart the app or not.

**Review by vusters — CHANGES_REQUESTED** [review]
Please read my original comments before you implement this.

File: lib/ensemble.dart [hunk 3]
```diff
@@ -46,6 +47,23 @@ class Ensemble {
     externalScreenWidgets = widgets;
   }
 
+  final List<Map<String, Object?>> _afterInitMethods = [];
+  void addCallbackAfterInitialization(
+      {required Function method,
+      List<dynamic>? positionalArgs,
+      Map<String, dynamic>? namedArgs}) {
+    final function = {
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/ensemble.dart:55]: @vusters 
I tried changing it to the list of functions, but how can I pass a method with an argument to the ```addCallbackAfterInitialization(method: )``` method?

Ex: 
```dart
Ensemble()
        .addCallbackAfterInitialization(method: navigateToScreen(URL));
```

**Review by vusters — APPROVED** [review]

**Pull request merged**